### PR TITLE
show more link results 

### LINKED ISSUE #150377: Allow terminal link detection commands to read links from more than just the viewport
Problem reported via https://github.com/microsoft/vscode/issues/149882#issuecomment-1137054006

This would be helpful especially for screen reader users.

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #150377

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts [hunk 1]
```diff
@@ -141,12 +143,20 @@ export class TerminalLinkManager extends DisposableStore {
...
-		for (let i = this._xterm.buffer.active.length - 1; i >= this._xterm.buffer.active.viewportY; i--) {
+		let noMoreResults: boolean = false;
+		let topLine = !extended ? this._xterm.buffer.active.viewportY - Math.min(this._xterm.rows, 50) : this._lastTopLine! - (Math.min(this._xterm.rows, 50));
+		if (topLine < 0 || topLine - Math.min(this._xterm.rows, 50) < 0) {
+			noMoreResults = true;
+		}
+		if (topLine < 0) {
+			topLine = 0;
+		}
+		this._lastTopLine = topLine;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts:158]: This looks like fixed size paging, can we do something more aggressive here like pull in 1000 per page or more if performance seems fine?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts:158]: I worry that pulling in a non-standard number might be surprising/unexpected from an accessibility perspective. I think it makes sense to have fixed size paging here.

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts:158]: will add a show all

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts:158]: will make the page 1000

**Review by Tyriar — APPROVED** [review]

**Pull request merged**